script/sync_prod_to_gcp/submissions_to_gcp.py

```diff
@@ -0,0 +1,928 @@
+            subject = f"Uploading of {paper_id}v{version} failed"
+            mail_body = f"Hello EUST,\nSubmission uploading for {paper_id}v{version} has failed. Please resolve the issue.\n\nThis message is generated by a bot on arxiv-sync.serverfarm.cornell.edu.\n"
+            cmd = ["/usr/bin/mail", "-r", "developers@arxiv.org", "-s", subject, help_needed]
+            mail = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            try:
+                mail.communicate(mail_body, timeout=60)
+                if mail.returncode == 0:
+                    # Once the email message is sent, done here.
+                    message.ack()
+                    logger.warning(f"Alart mail sent: {subject}", extra=log_extra)
+                    return
+                else:
+                    logger.error("Failed to send mail: %s", shlex.join(cmd), extra=log_extra)
+            except Exception as exc:
+                logger.error(f"Failed: %s", shlex.join(cmd), extra=log_extra, exc_info=True)
+                pass
+            pass
+
+        message.nack()
+        return
+
+    except Exception as _exc:
+        logger.warning(
+            f"Unknown error xid: {paper_id}, mid: {str(message.message_id)}",
+            extra=log_extra, exc_info=True, stack_info=False)
+        message.nack()
+        return
+
+    try:
+        sync_to_gcp(state, log_extra)
+        message.ack()
+        # Acknowledge the message so it is not re-sent
+        logger.info("ack message: %s", state.xid.ids, extra=log_extra)
+
+    except Exception as _exc:
+        logger.error("Error processing message: {exc}", exc_info=True, extra=log_extra)
+        message.nack()
+        pass
+
+
+
+def sync_to_gcp(state: SubmissionFilesState, log_extra: dict) -> bool:
+    """
+    From the submission file state, copy the ones that should bu uploaded.
+
+    If the /ftp's submission contains extra files, rename them with /trash prefix so the
+    object is no longer under /ftp but under /trash/ftp.
+    """
+    gs_client = thread_data.storage
+    desired_state = state.get_expected_files()
+    xid = state.xid
+    log_extra["arxiv_id"] = xid.ids
+    logger.info("Processing %s", xid.ids, extra=log_extra)
+
+    # Existing blobs on GCP
+    cit_source_root_path = state.source_path("")
+    bucket_objects: typing.List[str] = \
+        list_bucket_objects(gs_client, path_to_bucket_key(cit_source_root_path))
+
+    # Objects synced to GCP
+    for entry in desired_state:
+        local = entry["cit"]
+        remote = entry["gcp"]
+        entry_type = entry['type']
+
+        if entry_type == "html-cache":
+            # "html-cache" is needed on CIT as it is the root dir of HTML submission, but
+            # it is NOT a bucket object, and nothing to upload. Subsequent object copy becomes
+            # effectively creating the "directory" on GCP.
+            logger.debug("Skipping [%s]: %s -> %s", entry_type, local, remote, extra=log_extra)
```

Comment:
I'm concerned that html source papers are not getting synced. For each html paper what is needed:
1. build the html at a CIT web node by requesting the html
2.sync the html from ps_cache to GCP

Reply:
I decided to remove the condition of checking to see the HTML root dir exists. I think that was a bug that if someone else asks the HTML first (or say before the submission gets to here), it may not have populated the html files. With the change, it should scoop up the files regardless of non-existance of HTML root dir, and tries to send the files. So thanks for making me think this.